src/zmsg.c

```diff
@@ -701,7 +701,10 @@ zmsg_decode (zframe_t *frame)
             break;
         }
         zframe_t *decoded = zframe_new (source, frame_size);
-        assert (decoded);
+        if (!decoded) {
```

Comment:
in what cases can this happen?

Reply:
This happens if frame_size causes a heap exhaustion which makes zframe_new return NULL (see zframe.c).

Reply:
we don't really support running when there's no memory available, it will break in millions of other ways

Reply:
We found this one by trying to decode data corresponding to a PNG image. This first bytes normally determine the size of the zframe but in this case, it created a heap exhaustion due to the decoded size. It's not strictly heap exhaustion but it may happen when data to be decoded is corrupted or not an encoded zframe.